src/js/client/editprefset-settingsTable-widget.js

```diff
@@ -0,0 +1,324 @@
+            args: ["{that}"]
+        },
+        // Overriding renderInitialMarkup to take `that` as a render context.
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}"]
+        },
+        updateLunrIndex: {
+            funcName: "gpii.devpmt.settingsTable.updateLunrIndex",
+            args: ["{that}"]
+        },
+        filterSettings: {
+            funcName: "gpii.devpmt.settingsTable.filterSettings",
+            args: ["{that}", "{that}.dom.settingsRows", "{that}.lunrIndex", "{that}.model.settingsFilter"]
+        },
+        enableProductListener: {
+            funcName: "gpii.devpmt.settingsTable.enableProductListener",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"] // event
+        },
+        updateTermUsage: {
+            funcName: "gpii.devpmt.settingsTable.updateTermUsage",
+            args: ["{that}"]
+        },
+        removeProduct: {
+            funcName: "gpii.devpmt.settingsTable.removeProduct",
+            args: ["{that}", "{gpii.devpmt.editPrefs}"]
+        }
+    },
+    markupEventBindings: {
+        enabledBooleanInputs: {
+            method: "click",
+            args: "{that}.enableProductListener"
+        },
+        addContextButton: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.events.openAddContextDialog.fire"
+        },
+        valueDisplayCell: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editValueEvent"
+        },
+        removeProductButton: {
+            method: "click",
+            args: "{that}.removeProduct"
+        },
+        pspShowCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspShow"
+        },
+        pspMemoryCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspMemory"
+        }
+    },
+    listeners: {
+        "onCreate": [
+            {
+                "func": "{that}.updateLunrIndex"
+            },
+            {
+                "func": "{that}.filterInit"
+            },
+            {
+                "func": "{that}.updateTermUsage"
+            }
+        ],
+        "onMarkupRendered": [
+            {
+                "func": "{that}.filterSettings"
+            }
+        ]
+    },
+    lunrIndex: null
+});
+
+/**
+ * gpii.devpmt.settingsTable.filterInit - The users current filter options
+ * will be stored on the parent component and updated via model relay, such
+ * that they are saved if this component is destroyed and recreated.
+ * The first time it is created though, we want to set the default values,
+ * and trigger them via change applier so they are set on the parent.
+ *
+ * @param (Object) that
+ */
+gpii.devpmt.settingsTable.filterInit = function (that) {
+    if (that.model.settingsFilter === null) {
+        that.applier.change("settingsFilter", "");
```

Comment:
Would the logic be clearer to set the initial model value for "settingsFilter" to "" (empty string) at line 28 while having an "onCreate" listener to call "{that}.filterSettings" for the initial filtering?

The same comment applies to the "allSettingsEnabled" handling at line 187-189.

Reply:
Thanks for thinking through this, after reviewing the way this works, the filter init isn't really necessary at all, the values can just be set on the incoming dynamic component modelRelay. This method has been removed.